alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1340 @@
+                if not np.all(mask):
+                    index = np.where(not mask)[0][0]
+                    raise ValueError(message.format(f, grid_points[f][index]))
+
+    def _features_sanity_checks(self, features: Optional[List[Union[int, Tuple[int, int]]]]) -> None:
+        """
+        Features sanity checks.
+
+        Parameters
+        ----------
+        features
+            List of feature indices or pairs of feature indices to compute the partial dependence for.
+        """
+        if features is None:
+            return
+
+        def check_feature(f):
+            if not isinstance(f, numbers.Integral):
+                raise ValueError(f'All feature entries must be integers. Got a feature value of {type(f)} type.')
+            if f >= len(self.feature_names):
+                raise ValueError(f'All feature entries must be less than len(feature_names)={len(self.feature_names)}. '
+                                 f'Got a feature value of {f}.')
+            if f < 0:
+                raise ValueError(f'All feature entries must be greater or equal to 0. Got a feature value of {f}.')
+
+        for f in features:
+            if isinstance(f, tuple):
+                if len(f) != 2:
+                    raise ValueError(f'Current implementation of the partial dependence supports a maximum of two '
+                                     f'features at a time when a tuple is passed. Received {len(f)} features with the '
+                                     f'values {f}.')
+
+                check_feature(f[0])
+                check_feature(f[1])
+            else:
+                check_feature(f)
+
+    def _params_sanity_checks(self,
+                              estimator: BaseEstimator,
+                              response_method: Literal['auto', 'predict_proba', 'decision_function'] = 'auto',
+                              method: Literal['auto', 'recursion', 'brute'] = 'auto',
+                              kind: Literal['average', 'individual', 'both'] = 'average'
+                              ) -> Tuple[str, str, str]:
+        """
+        Parameters sanity checks. Most of the code is borrowed from:
+        https://github.com/scikit-learn/scikit-learn/blob/baf0ea25d/sklearn/inspection/_partial_dependence.py
+
+        estimator
+            A `sklearn` estimator.
+        response_method, method, kind
+            See :py:meth:`alibi.explainers.partial_dependence.PartialDependence.explain` method.
+
+        Returns
+        -------
+        Update parameters `(response_method, method, kind)`.
+        """
+        check_is_fitted(estimator)
+
+        if not (is_classifier(estimator) or is_regressor(estimator)):
+            raise ValueError('The predictor must be a fitted regressor or a fitted classifier.')
+
+        if is_classifier(estimator) and isinstance(estimator.classes_[0], np.ndarray):
+            raise ValueError('Multiclass-multioutput predictors are not supported.')
```

Comment:
Would like to see these in a separate validation method to be called in the constructor so we can fail quickly if the estimator isn't fitted/supported.